tests/bsim/bluetooth/host/id/settings/test_scripts/settings.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/env bash
+# Copyright 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
+
+test_name='settings'
+test_exe="bs_${BOARD}_tests_bsim_bluetooth_host_id_${test_name}_prj_conf"
+simulation_id="id_${test_name}"
+verbosity_level=2
+EXECUTE_TIMEOUT=30
+
+cd ${BSIM_OUT_PATH}/bin
+
+if [ "${1}" != 'debug1' ]; then
+  Execute "./${test_exe}" \
+    -v=${verbosity_level} -s=${simulation_id} -d=0 -testid=dut1 \
+    -flash="${simulation_id}.log.bin"
+fi
+
+if [ "${1}" != 'debug2' ]; then
+  Execute "./${test_exe}" \
+    -v=${verbosity_level} -s=${simulation_id} -d=1 -testid=dut2 \
+    -flash="${simulation_id}.log.bin" -flash_rm
+fi
+
+Execute ./bs_2G4_phy_v1 -v=${verbosity_level} -s=${simulation_id} \
+  -D=2 -sim_length=60e6
+
+if [ "${1}" == 'debug1' ]; then
```

Comment:
I will not reject the PR based on this, this is just a comment:
It seems like if you are debugging, and you are ok with setting a parameter for the script, you would probably be ok with just modifying this line in the script anyhow.

You can expect others will run the debugger from the IDE directly or in other ways, and that they will not expect about this option so adding all this `if != / == debug` does not really seem to add value for others, as much as be left overs from development. => I would remove them.